lib/request.coffee

```diff
@@ -97,6 +103,32 @@ module.exports = getRequest = ({ token, debug = false, retries = 0, isBrowser =
 
 			return options
 
+	interceptRequestOptions = (requestOptions) ->
+		interceptRequestOrError(Promise.resolve(requestOptions))
+
+	interceptRequestError = (requestError) ->
+		interceptRequestOrError(Promise.reject(requestError))
+
+	interceptResponse = (response) ->
+		interceptResponseOrError(Promise.resolve(response))
+
+	interceptResponseError = (responseError) ->
+		interceptResponseOrError(Promise.reject(responseError))
+
+	interceptRequestOrError = (initialPromise) ->
+		Promise.resolve(
+			exports.interceptors.reduce (promise, interceptor) ->
+				promise.then(interceptor.request, interceptor.requestError)
+			, initialPromise
+		)
+
+	interceptResponseOrError = (initialPromise) ->
+		Promise.resolve(
+			exports.interceptors.slice().reverse().reduce (promise, interceptor) ->
+				promise.then(interceptor.response, interceptor.responseError)
```

Comment:
if both are null-ish will `p.then(null, null)` resolve with the original promise value?